File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if (root == null){
            return res;
        }

        Queue<TreeNode> lvl = new LinkedList<>();
        lvl.add(root);

        while (!lvl.isEmpty()){
            List<Integer> nodes = new ArrayList<>();
            int iterate = lvl.size();
            for (int i = 0; i < iterate; i++){
                TreeNode node = lvl.remove();
                nodes.add(node.val); // FIFO
                if (node.left != null){
                    lvl.add(node.left);
                }
                if (node.right != null){
                    lvl.add(node.right);
                }
            }
            res.add(nodes);
        }

        return res;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0
        def dfs(root, maxSeen=-float('infinity')):
            if root:
                if root.val >= maxSeen:
                    self.res += 1
                    maxSeen = root.val
                dfs(root.left, maxSeen)
                dfs(root.right, maxSeen)
        dfs(root)
        return self.res